### PR TITLE
kraken: add metric for request currently handled

### PR DESCRIPTION
**Pull request by kinnou02**

### CODE REVIEW

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Review by nonifier — APPROVED** [review]

**Comment by kinnou02** [comment]
Jenkins, test this please.

File: source/kraken/metrics.cpp [hunk 1]
```diff
@@ -72,6 +84,20 @@ Metrics::Metrics(const boost::optional<std::string>& endpoint, const std::string
...
+                        .Register(*registry);
+    this->in_flight = &in_flight_family.Add({});
+}
+
+std::unique_ptr<InFlightGuard> Metrics::start_in_flight() const{
+    if(!registry) {
+        return std::make_unique<InFlightGuard>(nullptr);
+    }
+    return std::make_unique<InFlightGuard>(this->in_flight);
+
```
Comment (pbougue) [inline source/kraken/metrics.cpp]: minor: useless extra line

File: source/kraken/metrics.h [hunk 2]
```diff
@@ -49,15 +50,27 @@ class Histogram;
...
     std::shared_ptr<prometheus::Registry> registry;
     std::map<pbnavitia::API, prometheus::Histogram*> request_histogram;
+    prometheus::Gauge* in_flight;
 public:
     Metrics(const boost::optional<std::string>& endpoint, const std::string& coverage);
     void observe_api(pbnavitia::API api, double duration) const;
+    std::unique_ptr<InFlightGuard> start_in_flight() const;
 };
 
+
```
Comment (pbougue) [inline source/kraken/metrics.h]: minor: useless extra line

File: source/kraken/metrics.h [hunk 3]
```diff
@@ -49,15 +50,27 @@ class Histogram;
...
 namespace navitia {
-class Metrics: boost::noncopyable {
 
+class InFlightGuard: boost::noncopyable {
+    prometheus::Gauge* gauge;
+public:
+    InFlightGuard(prometheus::Gauge* gauge);
+    ~InFlightGuard();
+};
+
```
Comment (pbougue) [inline source/kraken/metrics.h:64]: minor: useless extra line

**Review by pbougue — APPROVED** [review]

File: source/kraken/kraken_zmq.h [hunk 4]
```diff
@@ -91,6 +91,7 @@ inline void doWork(zmq::context_t& context,
             continue;
         }
 
+        auto in_flight_guard = metrics.start_in_flight();
```
Comment (xlqian) [inline source/kraken/kraken_zmq.h]: minor: I think you can construct an object InFlightGuard here without unique_ptr?

**Review by xlqian — APPROVED** [review]

File: source/kraken/kraken_zmq.h (hunk 4 above)
Comment (kinnou02) [reply source/kraken/kraken_zmq.h]: should be better now

**Review by xlqian — APPROVED** [review]

**Pull request merged**